src/Evolution/Initialization/Evolution.hpp

```diff
@@ -137,8 +137,8 @@ struct Evolution {
         initial_time);
 
     return db::create_from<db::RemoveTags<>, simple_tags, compute_tags>(
-        std::move(box), TimeId{}, time_id, initial_dt, std::move(dt_vars),
-        std::move(history));
+        std::move(box), TimeId{time_runs_forward, 0, initial_time}, time_id,
```

Comment:
What is this change about?  The previous behavior was intentional.  (And if there is a reason to set this this is not the correct value to set it to.  It should be `time_id`.)

Reply:
Registration with observers registers the temporal id and so that needs to be valid. Without this I cannot register with observers. One option could be to do registration after setting the time history, which would just mean that we cannot observe during initialization of the time history. I don't expect as to typically want to do this, but it could be useful for debugging. I don't have a preference here :)

Reply:
Apparently the value doesn't matter, so can you just hardcode something in the registration code?  Observers using this to register is probably wrong anyway (#1534).

Reply:
A couple questions to make sure I understand:
- Will we need to worry about potentially have roundoff error differences in the time double between nodes? This could cause issues for the observers (I think that's already a possible concern with the current code).
- How would we register and deregister during load balancing if we hard code something into registration? A future problem, but that's why I don't hard code any value.

Reply:
* Re roundoff:
  * For dense output stuff the values used will be whatever is supplied to the dense trigger manager.  For values directly specified by the user these will obviously agree across all elements.  For other cases making sure everything agrees is the job of whatever requests the trigger, but in practice it will probably be a broadcast value or something like a slab boundary.
  * Every element should agree on the value of a big-T Time because there is only one code path that calculates the value.  (Whether this is guaranteed between binaries built with different compilers or compiler options I am less certain of.  Short of unsafe math operations it should be fine.  (Only the basic arithmetic operations are used.))  Slab boundaries are trickier, since they are fundamentally floating-point, but they are calculated using only addition and subtraction, so there shouldn't be much opportunity for problematic math there.  (And again there is only one code path outside the tests that does it.)  (We do need to watch out for things assuming stuff like `slab.advance().retreat() == slab`, but currently slabs are never calculated backwards compared to the simulation flow and hopefully we can continue to avoid doing that.)  Slab size changing may also have subtleties, but that doesn't exist yet.  A lot of the low-level Time functions should really be moved out of header files, both to reduce the likelihood of differing math optimizations between compilation units and just to comply with modern style.  (The critical one is already in the cpp file.)
* Re registration:
  * I don't remember how registration works, but it is currently being passed the wrong value and working, so it can't care too much about what the value is.

Reply:
- Re roundoff:
  Okay, that sounds like it shouldn't be an issue then since all elements will agree on the `double` time. I think this is what we're currently using anyway.
- Re registration:
  * That's right, the code currently doesn't care, but as soon as we do load balancing it will care because which elements will send the observer data is time dependent. If you think we should punt on this issue I'm fine with hard-coding a dummy time for now during registration. We will need to address this issue soonish, though so it would be good to start thinking about :)